phalcon/http/message/request.zep

```diff
@@ -0,0 +1,689 @@
+	{
+		if (typeof name !== "string" ||
+			!preg_match("/^[a-zA-Z0-9\'`#$%&*+.^_|~!-]+$/", name)) {
+			throw new \InvalidArgumentException("Invalid header name " . name);
+		}
+	}
+
+	/**
+	 * Validates a header value
+	 *
+	 * Most HTTP header field values are defined using common syntax
+	 * components (token, quoted-string, and comment) separated by
+	 * whitespace or specific delimiting characters.  Delimiters are chosen
+	 * from the set of US-ASCII visual characters not allowed in a token
+	 * (DQUOTE and "(),/:;<=>?@[\]{}").
+	 *
+	 *     token          = 1*tchar
+	 *
+	 *     tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
+	 *                    / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
+	 *                    / DIGIT / ALPHA
+	 *                    ; any VCHAR, except delimiters
+	 *
+	 * A string of text is parsed as a single value if it is quoted using
+	 * double-quote marks.
+	 *
+	 *     quoted-string  = DQUOTE *( qdtext / quoted-pair ) DQUOTE
+	 *     qdtext         = HTAB / SP /%x21 / %x23-5B / %x5D-7E / obs-text
+	 *     obs-text       = %x80-FF
+	 *
+	 * Comments can be included in some HTTP header fields by surrounding
+	 * the comment text with parentheses.  Comments are only allowed in
+	 * fields containing "comment" as part of their field value definition.
+	 *
+	 *     comment        = "(" *( ctext / quoted-pair / comment ) ")"
+	 *     ctext          = HTAB / SP / %x21-27 / %x2A-5B / %x5D-7E / obs-text
+	 *
+	 * The backslash octet ("\") can be used as a single-octet quoting
+	 * mechanism within quoted-string and comment constructs.  Recipients
+	 * that process the value of a quoted-string MUST handle a quoted-pair
+	 * as if it were replaced by the octet following the backslash.
+	 *
+	 *     quoted-pair    = "\" ( HTAB / SP / VCHAR / obs-text )
+	 *
+	 * A sender SHOULD NOT generate a quoted-pair in a quoted-string except
+	 * where necessary to quote DQUOTE and backslash octets occurring within
+	 * that string.  A sender SHOULD NOT generate a quoted-pair in a comment
+	 * except where necessary to quote parentheses ["(" and ")"] and
+	 * backslash octets occurring within that comment.
+	 */
+	private function checkHeaderValue(var value) -> void
+	{
+		if typeof value !== "string" && typeof value !== "int" && typeof value !== "float" {
+			throw new \InvalidArgumentException("Invalid header value");
+		}
+
+		let value = (string) value;
+
+		if (1 === preg_match("#(?:(?:(?<!\r)\n)|(?:\r(?!\n))|(?:\r\n(?![ \t])))#", value) ||
+			1 === preg_match("/[^\x09\x0a\x0d\x20-\x7E\x80-\xFE]/", value)) {
+			throw new \InvalidArgumentException("Invalid header value");
+		}
+	}
+
+	/**
+	 * Check the method
+	 */
+	private function checkMethod(var method = "") ->  <Request>
+	{
+    	array methods;
+
+    	let methods = [
+			"GET"     : 1,
+			"CONNECT" : 1,
+			"DELETE"  : 1,
+			"HEAD"    : 1,
+			"OPTIONS" : 1,
+			"POST"    : 1,
+			"PUT"     : 1,
+			"TRACE"   : 1
```

Comment:
Please add "PURGE" (Squid and Varnish support)

Reply:
Done

Reply:
Not done :)